torch/_dynamo/variables/torch.py

```diff
@@ -911,6 +911,53 @@ def speculate_subgraph(f, sub_args, graph_checkpoint, checkpoint):
                 comparable_state,
             )
 
+        # See NOTE [HigherOrderOperator tracing design] for more details.
+        def speculate_subgraph(f, sub_args, graph_checkpoint, checkpoint):
+            try:
+                with tx.output.new_subtracer() as tracer:
+                    args = []
+                    # One argument to graph per sub_args
+                    for a in sub_args:
+                        if isinstance(a, TensorVariable):
+                            tracer.create_graph_input(a.as_proxy().node.name)
+                            args.append(a)
+                        else:
+                            # call_function() needs a TensorVariable, therefore we construct
+                            # one with inner graph proxy.
+                            assert isinstance(a, torch.Tensor)
+                            proxy = tracer.create_graph_input("arg")
+                            args.append(
+                                wrap_fx_proxy(tx=tx, proxy=proxy, example_value=a)
+                            )
```

Comment:
nit: Move this assert to the top of the for loop, and give it a message about only supporting tensor or TensorVariable.

Also, you can potentially rewrite it a little.
```suggestion
            for a in sub_args:
                assert isinstance(a, (TensorVariable, torch.Tensor)), "Only Tensor/Tensor VariableTracker allowed here."
                
                if isinstance(a, torch.Tensor):
                    # call_function() needs a TensorVariable, therefore we construct
                    # one with inner graph proxy.
                    proxy = tracer.create_graph_input("arg")
                    a = wrap_fx_proxy(tx=tx, proxy=proxy, example_value=a)
                    assert isinstance(a, TensorVariable)
                else:
                    tracer.create_graph_input(a.as_proxy().node.name)

                args.append(a)
```

Reply:
SGTM, I was copy-pasting the original `speculate_subgraph`